src/components/Pressable.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { GestureObjects as Gesture } from '../handlers/gestures/gestureObjects';
+import { GestureDetector } from '../handlers/gestures/GestureDetector';
+import {
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+  TouchData,
+} from '../handlers/gestureHandlerCommon';
+import { PressEvent, PressableEvent, PressableProps } from './PressableProps';
+import { HoverGestureHandlerEventPayload } from '../handlers/gestures/hoverGesture';
+import { LongPressGestureHandlerEventPayload } from '../handlers/LongPressGestureHandler';
+import { Insets, StyleProp, View, ViewStyle, processColor } from 'react-native';
+import RNButton from './GestureHandlerButton';
+
+const DEFAULT_LONG_PRESS_DURATION = 500;
+const DEFAULT_HOVER_DELAY = 0;
+
+const numberAsInset = (value: number): Insets => ({
+  left: value,
+  right: value,
+  top: value,
+  bottom: value,
+});
+
+const addInsets = (a: Insets, b: Insets): Insets => ({
+  left: (a.left ?? 0) + (b.left ?? 0),
+  right: (a.right ?? 0) + (b.right ?? 0),
+  top: (a.top ?? 0) + (b.top ?? 0),
+  bottom: (a.bottom ?? 0) + (b.bottom ?? 0),
+});
+
+const touchToPressEvent = (data: TouchData, timestamp: number): PressEvent => ({
+  identifier: data.id,
+  locationX: data.x,
+  locationY: data.y,
+  pageX: data.absoluteX,
+  pageY: data.absoluteY,
+  target: 0, // fixme if possible, set to correct target ID
+  timestamp: timestamp,
+  touches: [], // intentionally empty
+  changedTouches: [], // intentionally empty
+});
+
+const changeToTouchData = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): TouchData => ({
+  id: 0, // fixme if possible, set to correct pointer ID
```

Comment:
Bump, to not forget about it

Reply:
Could not find a conventional way of getting `targetId`, and `pointerId` in those places that they are missing.